### ISSUE TITLE
`units` error in vignette chunk ¨elevation_profile"

### ISSUE THREAD

**Issue by patperu** [GIScience/openrouteservice-r#57]
Hi,

I'm getting an error running the chunk [elevation_profile](https://github.com/GIScience/openrouteservice-r/blob/master/vignettes/openrouteservice.Rmd#L190) from the vignette

    Fehler in c.units(c(0, 0.0193202875860916, 0.0255554983954827, 0.0408713583223733,  : 
       units are not convertible, and cannot be mixed; try setting units_options(allow_mixed = TRUE)?

Setting `units_options(allow_mixed = TRUE)` I get

     Fehler in UseMethod("mixed_units") : 
        nicht anwendbare Methode für 'mixed_units' auf Objekt der Klasse "logical" angewendet

Thanks for the package!
 
Patrick

**Comment by aoles** [comment]
Dear Patrick,

thanks a lot for the bug report! It seems to be related to some updates in the package dependencies, could you maybe paste the output of `sessionInfo()`?

Cheers,
Andrzej

**Comment by patperu** [comment]
Hello Andrzej,

sure - this is the output


```r
sessionInfo()
R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.3 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.7.1

locale:
 [1] LC_CTYPE=de_DE.UTF-8       LC_NUMERIC=C               LC_TIME=de_DE.UTF-8       
 [4] LC_COLLATE=de_DE.UTF-8     LC_MONETARY=de_DE.UTF-8    LC_MESSAGES=de_DE.UTF-8   
 [7] LC_PAPER=de_DE.UTF-8       LC_NAME=C                  LC_ADDRESS=C              
[10] LC_TELEPHONE=C             LC_MEASUREMENT=de_DE.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] units_0.6-4            ggforce_0.3.1          ggplot2_3.2.1          sf_0.8-1              
[5] googlePolylines_0.7.2  leaflet_2.0.2          openrouteservice_0.3.0

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.2         lattice_0.20-38    ps_1.3.0           class_7.3-15       assertthat_0.2.1  
 [6] digest_0.6.21      packrat_0.5.0      V8_2.3             mime_0.7           R6_2.4.0          
[11] reprex_0.3.0.9000  evaluate_0.14      e1071_1.7-2        httr_1.4.1.9000    geojson_0.3.2     
[16] pillar_1.4.2       rlang_0.4.0.9003   lazyeval_0.2.2     curl_4.2           geojsonlint_0.3.0 
[21] rstudioapi_0.10    callr_3.3.2        whisker_0.4        rmarkdown_1.16     jqr_1.1.0         
[26] htmlwidgets_1.5    polyclip_1.10-0    munsell_0.5.0      shiny_1.3.2.9001   compiler_3.6.1    
[31] httpuv_1.5.2.9000  xfun_0.10          pkgconfig_2.0.3    clipr_0.7.0        htmltools_0.4.0   
[36] tidyselect_0.2.5   tibble_2.1.3       httpcode_0.2.0     jsonvalidate_1.2.2 crayon_1.3.4      
[41] dplyr_0.8.3        withr_2.1.2        later_1.0.0        MASS_7.3-51.4      crul_0.8.4.9100   
[46] grid_3.6.1         jsonlite_1.6       lwgeom_0.1-7       xtable_1.8-4       gtable_0.3.0      
[51] DBI_1.0.0          magrittr_1.5       scales_1.0.0.9000  KernSmooth_2.23-15 cli_1.1.0         
[56] farver_1.1.0       fs_1.3.1           promises_1.1.0     sp_1.3-1           xml2_1.2.2        
[61] keyring_1.1.0      RColorBrewer_1.1-2 geojsonsf_1.3.2    tools_3.6.1        glue_1.3.1        
[66] tweenr_1.0.1       purrr_0.3.2        crosstalk_1.0.0    processx_3.4.1     fastmap_1.0.0     
[71] yaml_2.2.0         colorspace_1.4-1   sessioninfo_1.1.1  classInt_0.4-1     knitr_1.25 
```

And this from `sessioninfo::session_info() ` - I like it because you get the build dates and sources. I tend to use the most recent versions from github....

```r
sessioninfo::session_info() 
─ Session info ───────────────────────────────────────────────────────────────────────────────────
 setting  value                       
 version  R version 3.6.1 (2019-07-05)
 os       Ubuntu 18.04.3 LTS          
 system   x86_64, linux-gnu           
 ui       RStudio                     
 language (EN)                        
 collate  de_DE.UTF-8                 
 ctype    de_DE.UTF-8                 
 tz       Europe/Berlin               
 date     2019-10-09                  

─ Packages ───────────────────────────────────────────────────────────────────────────────────────
 package          * version    date       lib source                                
 assertthat         0.2.1      2019-03-21 [1] CRAN (R 3.6.1)                        
 callr              3.3.2      2019-09-22 [1] CRAN (R 3.6.1)                        
 class              7.3-15     2019-01-01 [4] CRAN (R 3.5.2)                        
 classInt           0.4-1      2019-08-06 [1] CRAN (R 3.6.1)                        
 cli                1.1.0      2019-03-19 [1] CRAN (R 3.6.1)                        
 clipr              0.7.0      2019-07-23 [1] CRAN (R 3.6.1)                        
 colorspace         1.4-1      2019-03-18 [1] CRAN (R 3.6.1)                        
 crayon             1.3.4      2017-09-16 [1] CRAN (R 3.6.1)                        
 crosstalk          1.0.0      2016-12-21 [1] CRAN (R 3.6.1)                        
 crul               0.8.4.9100 2019-08-28 [1] Github (ropensci/crul@2338b55)        
 curl               4.2        2019-09-24 [1] CRAN (R 3.6.1)                        
 DBI                1.0.0      2018-05-02 [1] CRAN (R 3.6.1)                        
 digest             0.6.21     2019-09-20 [1] CRAN (R 3.6.1)                        
 dplyr              0.8.3      2019-07-04 [1] CRAN (R 3.6.1)                        
 e1071              1.7-2      2019-06-05 [1] CRAN (R 3.6.1)                        
 evaluate           0.14       2019-05-28 [1] CRAN (R 3.6.1)                        
 farver             1.1.0      2018-11-20 [1] CRAN (R 3.6.1)                        
 fastmap            1.0.0      2019-07-28 [1] CRAN (R 3.6.1)                        
 fs                 1.3.1      2019-05-06 [1] CRAN (R 3.6.1)                        
 geojson            0.3.2      2019-01-31 [1] CRAN (R 3.6.1)                        
 geojsonlint        0.3.0      2019-02-08 [1] CRAN (R 3.6.1)                        
 geojsonsf          1.3.2      2019-10-08 [1] Github (SymbolixAU/geojsonsf@8180d83) 
 ggforce          * 0.3.1      2019-08-20 [1] CRAN (R 3.6.1)                        
 ggplot2          * 3.2.1      2019-08-10 [1] CRAN (R 3.6.1)                        
 glue               1.3.1      2019-03-12 [1] CRAN (R 3.6.1)                        
 googlePolylines  * 0.7.2      2018-11-23 [1] CRAN (R 3.6.1)                        
 gtable             0.3.0      2019-03-25 [1] CRAN (R 3.6.1)                        
 htmltools          0.4.0      2019-10-04 [1] CRAN (R 3.6.1)                        
 htmlwidgets        1.5        2019-10-04 [1] CRAN (R 3.6.1)                        
 httpcode           0.2.0      2016-11-14 [1] CRAN (R 3.6.1)                        
 httpuv             1.5.2.9000 2019-10-03 [1] Github (rstudio/httpuv@57e1d71)       
 httr               1.4.1.9000 2019-08-27 [1] Github (r-lib/httr@6b84c57)           
 jqr                1.1.0      2018-10-22 [1] CRAN (R 3.6.1)                        
 jsonlite           1.6        2018-12-07 [1] CRAN (R 3.6.1)                        
 jsonvalidate       1.2.2      2019-10-08 [1] Github (ropensci/jsonvalidate@ab6c6fe)
 KernSmooth         2.23-15    2015-06-29 [4] CRAN (R 3.5.0)                        
 keyring            1.1.0      2018-07-16 [1] CRAN (R 3.6.1)                        
 knitr              1.25       2019-09-18 [1] CRAN (R 3.6.1)                        
 later              1.0.0      2019-10-04 [1] CRAN (R 3.6.1)                        
 lattice            0.20-38    2018-11-04 [4] CRAN (R 3.5.1)                        
 lazyeval           0.2.2      2019-03-15 [1] CRAN (R 3.6.1)                        
 leaflet          * 2.0.2      2018-08-27 [1] CRAN (R 3.6.1)                        
 lwgeom             0.1-7      2019-05-06 [1] CRAN (R 3.6.1)                        
 magrittr           1.5        2014-11-22 [1] CRAN (R 3.6.1)                        
 MASS               7.3-51.4   2019-04-26 [4] CRAN (R 3.6.1)                        
 mime               0.7        2019-06-11 [1] CRAN (R 3.6.1)                        
 munsell            0.5.0      2018-06-12 [1] CRAN (R 3.6.1)                        
 openrouteservice * 0.3.0      2019-10-09 [1] local                                 
 packrat            0.5.0      2018-11-14 [1] CRAN (R 3.6.1)                        
 pillar             1.4.2      2019-06-29 [1] CRAN (R 3.6.1)                        
 pkgconfig          2.0.3      2019-09-22 [1] CRAN (R 3.6.1)                        
 polyclip           1.10-0     2019-03-14 [1] CRAN (R 3.6.1)                        
 processx           3.4.1      2019-07-18 [1] CRAN (R 3.6.1)                        
 promises           1.1.0      2019-10-04 [1] CRAN (R 3.6.1)                        
 ps                 1.3.0      2018-12-21 [1] CRAN (R 3.6.1)                        
 purrr              0.3.2      2019-03-15 [1] CRAN (R 3.6.1)                        
 R6                 2.4.0      2019-02-14 [1] CRAN (R 3.6.1)                        
 RColorBrewer       1.1-2      2014-12-07 [1] CRAN (R 3.6.1)                        
 Rcpp               1.0.2      2019-07-25 [1] CRAN (R 3.6.1)                        
 reprex             0.3.0.9000 2019-08-28 [1] Github (tidyverse/reprex@f888a72)     
 rlang              0.4.0.9003 2019-10-08 [1] Github (r-lib/rlang@09fda4a)          
 rmarkdown          1.16       2019-10-01 [1] CRAN (R 3.6.1)                        
 rstudioapi         0.10       2019-03-19 [1] CRAN (R 3.6.1)                        
 scales             1.0.0.9000 2019-08-09 [1] Github (r-lib/scales@7f6f4a5)         
 sessioninfo        1.1.1      2018-11-05 [1] CRAN (R 3.6.1)                        
 sf               * 0.8-1      2019-09-28 [1] Github (r-spatial/sf@cf93fc1)         
 shiny              1.3.2.9001 2019-10-01 [1] Github (rstudio/shiny@89bd7e9)        
 sp                 1.3-1      2018-06-05 [1] CRAN (R 3.6.1)                        
 tibble             2.1.3      2019-06-06 [1] CRAN (R 3.6.1)                        
 tidyselect         0.2.5      2018-10-11 [1] CRAN (R 3.6.1)                        
 tweenr             1.0.1      2018-12-14 [1] CRAN (R 3.6.1)                        
 units            * 0.6-4      2019-08-22 [1] CRAN (R 3.6.1)                        
 V8                 2.3        2019-07-02 [1] CRAN (R 3.6.1)                        
 whisker            0.4        2019-08-28 [1] CRAN (R 3.6.1)                        
 withr              2.1.2      2018-03-15 [1] CRAN (R 3.6.1)                        
 xfun               0.10       2019-10-01 [1] CRAN (R 3.6.1)                        
 xml2               1.2.2      2019-08-09 [1] CRAN (R 3.6.1)                        
 xtable             1.8-4      2019-04-21 [1] CRAN (R 3.6.1)                        
 yaml               2.2.0      2018-07-25 [1] CRAN (R 3.6.1)                        

[1] /home/patrick/R/x86_64-pc-linux-gnu-library/3.6
[2] /usr/local/lib/R/site-library
[3] /usr/lib/R/site-library
[4] /usr/lib/R/library
```
I'll add this output to issue https://github.com/GIScience/openrouteservice-r/issues/58 as well.

Thanks for your work!

Cheers
Patrick

**Comment by aoles** [comment]
Thank you for providing the details!

Unfortunately, so far I was not able to reproduce either of the reported problems. The CI builds on Travis and AppVeyor are fine too. So I guess the observed issues must be related to your specific configuration, maybe to the devel versions of the packages you are using? I will put this on hold for the time being, please leave a note here in case you know more.

Cheers,
Andrzej

**Comment by aoles** [comment]
Dear Patrick,

Do you still observe the described problem? If not, I would suggest closing the issue for now - it can be always be re-opened if necessary.

Cheers,
Andrzej

**Comment by patperu** [comment]
Issue resolved, see #58
